src/lib/core.d.ts

```diff
@@ -836,7 +836,7 @@ interface RegExp {
     lastIndex: number;
 
     // Non-standard extensions
-    compile(): RegExp;
+    compile(): RegExp | undefined;

```

Comment:
From what I can tell this function is actually supposed to return only undefined since it updates the RegExp in-place. So it was wrong before and should be `: void`